### PR TITLE
Overview of Property Locking

### PR DESCRIPTION
**Pull request by greg-szrama**
Added overview of property locking feature

[Link](https://deploy-preview-123--redpanda-docs-preview.netlify.app/23.3/manage/cluster-maintenance/cluster-property-configuration/#lock-properties-at-the-cluster-level)

### CODE REVIEW

File: modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc [hunk 1]
```diff
@@ -68,6 +68,18 @@ Successfully updated configuration. New configuration version is 2.
 
 NOTE: You can also change property values using the `rpk cluster config set` command, but this method does not display the current setting or the description.
 
+== Lock Properties at the Cluster Level
+
+In many instances, cluster administrators wish to lock certain properties in a way that individual topic settings cannot override them. If a property is locked then any attempts to edit that property at a topic level fail. For example, if you lock `internal_topic_replication_factor` at a value of 3, then any attempt to set `replication.factor` for a topic will yield an error.
```
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]:
```suggestion
In many instances, cluster administrators may wish to lock certain properties in a way that individual topic settings cannot override them. If a property is locked, then any attempts to edit that property at a topic level will fail. For example, if you lock `internal_topic_replication_factor` at a value of 3, then any attempt to set or edit `replication.factor` for a topic will yield an error.
```

File: modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc [hunk 2]
```diff
@@ -68,6 +68,18 @@ Successfully updated configuration. New configuration version is 2.
...
+
+NOTE: No properties shipped with a locked setting. Property locking is a strictly opt-in feature.
```
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]:
```suggestion
NOTE: No properties are shipped with a locked setting. Property locking is a strictly opt-in feature.
```

File: modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc [hunk 3]
```diff
@@ -68,6 +68,18 @@ Successfully updated configuration. New configuration version is 2.
...
+
+Enabling a property lock on a topic uses a straightforward syntax - you append `;locked` to the property value. For example, if you run `rpk cluster config edit` you might set `log_compression_type=producer;locked`. When viewing the property, you will see the appended keyword giving a quick identification of the presence of the lock. Any YAML generated via `rpk cluster config export` also includes the presence of the locked attribute.
```
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]: Is it lock topic or locked topic? Seems like it should be locked topic, no?

Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]:
```suggestion
Enabling a property lock on a topic uses a straightforward syntax - you append `;locked` to the property value. For example, if you run `rpk cluster config edit` you might set `log_compression_type=producer;locked`. When viewing the property, you will see the appended keyword giving a quick identification of the presence of the lock. Any YAML generated using `rpk cluster config export` also indentifies any locked attributes.
```

File: modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc [hunk 4]
```diff
@@ -78,6 +90,8 @@ $ rpk cluster config get log_compression_type
 producer
 ----
 
+TIP: If you previously set a lock on the property, you will see the `;locked` attribute present in the output if this command.
```
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]:
```suggestion
TIP: If you previously set a lock on the property, you will see the `;locked` attribute present in the output of this command.
```

File: modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc [hunk 5]
```diff
@@ -68,6 +68,18 @@ Successfully updated configuration. New configuration version is 2.
...
+
+In many instances, cluster administrators wish to lock certain properties in a way that individual topic settings cannot override them. If a property is locked then any attempts to edit that property at a topic level fail. For example, if you lock `internal_topic_replication_factor` at a value of 3, then any attempt to set `replication.factor` for a topic will yield an error.
+
+NOTE: No properties shipped with a locked setting. Property locking is a strictly opt-in feature.
+
+Enabling a property lock on a topic uses a straightforward syntax - you append `;locked` to the property value. For example, if you run `rpk cluster config edit` you might set `log_compression_type=producer;locked`. When viewing the property, you will see the appended keyword giving a quick identification of the presence of the lock. Any YAML generated via `rpk cluster config export` also includes the presence of the locked attribute.
+
+NOTE: You can only set a property to locked when providing a new or existing value for the property. You cannot set a property to locked with a default value unless you explicitly specify that value along with the locked attribute.
```
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]: Not sure I follow the second statement here--so if a property has a default value you cannot lock it in? I had to read this a few times and I am still not clear what the behavior is here. Can you clarify? Thx.

Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]: That's correct, you can only lock it when setting it the the same or a new value. I'll try to rephrase this better.

File: modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc [hunk 6]
```diff
@@ -68,6 +68,24 @@ Successfully updated configuration. New configuration version is 2.
 
 NOTE: You can also change property values using the `rpk cluster config set` command, but this method does not display the current setting or the description.
 
+== Lock Property Values at the Cluster Level
```
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/cluster-property-configuration.adoc]:
```suggestion
== Lock property values at the cluster level
```

**Comment by Feediver1** [comment]
@greg-szrama Can you provide a status update here? Also, who is thee eng SME reviewer? thx

**Comment by greg-szrama** [comment]
Per Matt Schumpert:

SME will be @BenPope , but this feature is undergoing a bit of a redesign over the next weeks and will land late in 23.3 if it lands at all. Its not part of any beta builds at the moment

Target seems to be mid-december

**Comment by mattschumpert** [comment]
@greg-szrama has. this actually been implemented yet?  We should link to the redpanda/core-internal GH issue this is documented in the description please.

cc @BenPope  is this really done already?

**Comment by BenPope** [comment]
> cc @BenPope  is this really done already?

No, very much in progress.

**Comment by mattschumpert** [comment]
@greg-szrama where did you get this info.  So much has changed since the early days of this feature, and it's not even implemented yet. We should be holding off until its done.

**Review by mattschumpert — CHANGES_REQUESTED** [review]
Property locking does not exist yet. It's a feature very much still at risk for 23.3 and in flight. 

This PR should link to the issue it's resolving and the docs where the content is coming from.  Not familiar with this property syntax even.  

The origina RFC is here: https://docs.google.com/document/d/1M7ZhOA23hiW3RW6HjCEWhe64sWPXGssmkjyxDDDzNYM/edit#heading=h.rjq3le2j5m4a

**Comment by greg-szrama** [comment]
I based the changes on the PRD I received. The docs issue was:

https://github.com/redpanda-data/documentation-private/issues/1948

I was documenting this feature based on the above issue and the PRD that was attached to it. The syntax was specified in the docs issue. It sounds like I over-interpreted the completeness of the issue when documenting it based on that info.

**Comment by mattschumpert** [comment]
Thanks. No worries.  The RFC typically has the details of the implementation. that  doc issue had a paste of an old early draft of requirements (pre-PRD).   @BenPope will reach out when the feature is in a testable state to coordinate writing docs for what actually landed.

**Comment by mattschumpert** [comment]
@BenPope , @michael-redpanda 

Given this is slipping, not sure if we should keep this doc draft around (should be based on your RFC @BenPope ). DOes the RFC represent what its going to look like if we land it in 24.1?

**Comment by Feediver1** [comment]
Didn't land in time for 23.3 
Re-open when eng completes  @greg-szrama

**Comment by BenPope** [comment]
> @BenPope , @michael-redpanda 
> 
> Given this is slipping, not sure if we should keep this doc draft around (should be based on your RFC @BenPope ). DOes the RFC represent what its going to look like if we land it in 24.1?

No, it'll probably be quite different.

**Pull request closed without merging**